### PR TITLE
ADR: How to handle minigame configuration ID as URL route

### PR DESCRIPTION
**Pull request by madebyTimo**
please vote for your preference

### CODE REVIEW

**Comment by madebyTimo** [comment]
:+1: for insert the ID into the route

**Comment by madebyTimo** [comment]
:+1:  for keep the old solution (URL fragent)

**Review by maexled — APPROVED** [review]
insert the ID into the route

File: adr/2022-07-17 minigame-configuration-id-in-url.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- `/api/minigames/$MINIGAME/$ID/`
+    - easy to understand and use
+
+## Contra Solution
+- `/api/minigames/$MINIGAME/$ID/`
+    - changes needed in implementations of the old format
+- old solution
+    - very bad
+    - cannot use the fragment for section-linking anymore
+    - url fragment is normally optional but mandatory in our case
```
Comment (delvh) [inline adr/2022-07-17 minigame-configuration-id-in-url.md:27]:
```suggestion
    - URL fragment is normally optional but mandatory in our case
```

**Review by delvh — APPROVED** [review]

**Review by cdingm — APPROVED** [review]

**Comment by LeviUniAcc** [comment]
Do i get it correctly, we Change the # With a $?

**Comment by delvh** [comment]
> Do i get it correctly, we Change the # With a $?

Absolutely not.
The `$` is simply meant to signify that what follows is a variable (most common way to clarify that something is a variable, most likely adopted from POSIX shells where that syntax accesses said variable).
What we do instead is pass the config ID as a `path param` instead of as `URL fragment` (you might have already heard about path params/ worked with them in Spring i.e. for `PE2` or `Sopra`)

**Review by LeviUniAcc — APPROVED** [review]

**Review by Aaron2000119 — APPROVED** [review]

**Review by WuestFlorian — APPROVED** [review]

**Comment by delvh** [comment]
No activity here.
Merging now.

**Pull request merged**